charts/namada-indexer/templates/service-account.yaml

```diff
@@ -0,0 +1,19 @@
+{{- if .Values.serviceAccount.create -}}
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: {{ include "namada-indexer.serviceAccountName" . }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+    {{- include "namada-indexer.labels" . | nindent 4 }}
+  annotations:
+    {{- if .Values.argocd.deployment }}
+    "argocd.argoproj.io/sync-wave": "-4"
+    {{- else }}
+    "helm.sh/hook": "pre-install,pre-upgrade"
+    "helm.sh/hook-weight": "-10"
```

Comment:
Just a note: I believe argocd and helm hook annotations can co-exist without problems